app/gui2/src/components/TextFormattingSelector.vue

```diff
@@ -0,0 +1,132 @@
+<script setup lang="ts">
+import icons from '@/assets/icons.svg'
+import DropdownMenu from '@/components/DropdownMenu.vue'
+import SvgButton from '@/components/SvgButton.vue'
+import { ref } from 'vue'
+import { TextFormatOptions } from './visualizations/TableVisualization.vue'
+
+const emit = defineEmits<{
+  changeFormat: [formatValue: TextFormatOptions]
+}>()
+
+const textFormatterSelected = ref(TextFormatOptions.Partial)
+const setTextFormatterSelected = (option: TextFormatOptions) => {
+  textFormatterSelected.value = option
+  emit('changeFormat', textFormatterSelected.value)
+}
```

Comment:
Vue reactive values do not generally need setters.
```suggestion
watch(textFormatterSelected, (selected) => emit('changeFormat', selected))
```